### PR TITLE
Create a workflow to sync the list of authors

### PR DESCRIPTION
**Pull request by norm**
The CI jobs caused by Pull Requests run on Airflow infrastructure only when an author is trusted, which currently means a core committer. The list is currently duplicated in the ci.yml workflow.

As part of being able to open up the infra to other trusted authors, this list needs to be updated based on a list of trusted authors kept in the `airflow-ci-infra` repository.

### CODE REVIEW

**Review by dstandish — APPROVED** [review]

File: .github/workflows/sync_authors.yml
Comment (potiuk) [inline .github/workflows/sync_authors.yml:20]: we should explicitly list permissions here - this is far safer as we will only give the actual permissions needed by the workflow.

File: .github/workflows/sync_authors.yml
Comment (potiuk) [inline .github/workflows/sync_authors.yml]: We have the approach that "workflow_dispatch" workflows should also be limited to a few people - for example to release the prod images, we have a workflow limited to few release managers only - we probably should do something similar here.

**Review by potiuk — DISMISSED** [review]
Nice. I wonder how it wiill look like in reality :) -  permissions should be updaated but it looks sound.

File: .github/workflows/sync_authors.yml
Comment (norm) [reply .github/workflows/sync_authors.yml]: This was left in because I was testing the workflow in another repo first.

File: .github/workflows/sync_authors.yml
Comment (norm) [reply .github/workflows/sync_authors.yml:20]: This is why I had it in draft, I hadn't yet worked out what was needed there. :)

**Comment by norm** [comment]
@potiuk It should create a PR with a single line change (or more, if multiple people are added) to the `ci.yml`. If I've done it right :)

File: scripts/ci/runners/sync_authors.py
Comment (ashb) [inline scripts/ci/runners/sync_authors.py:57]: https://pypi.org/project/ruamel.yaml/ might be a slightly more maintable way than a regex. Though we'd still need to regex the value

**Review by ashb — APPROVED** [review]

File: scripts/ci/runners/sync_authors.py
Comment (norm) [reply scripts/ci/runners/sync_authors.py:57]: Updating the yaml with that does not preserve the contents of the file well, even with `RoundTripLoader/Dumper`. Most comments are stripped, and some parts are rewritten such that I would not be confident it would even work.

Comment (norm) [reply scripts/ci/runners/sync_authors.py:57]: eg

```
-      DEFAULT_CONSTRAINTS_BRANCH: ${{ needs.build-info.outputs.default-constraints-branch }}
+      DEFAULT_CONSTRAINTS_BRANCH: ${{ needs.build-info.outputs.default-constraints-branch
+        }}
```

Comment (norm) [reply scripts/ci/runners/sync_authors.py:57]: That particular breakage I've fixed — it wants a `width=180` or similar large number in order to not wrap.

Comment (ashb) [reply scripts/ci/runners/sync_authors.py:57]: Happy to merge this as is if you don't think ruamel.yaml is worth it (it may not be)

Comment (norm) [reply scripts/ci/runners/sync_authors.py:57]: I've tried all the options I can find and still can't get it to roundtrip without stripping out many of the comments. Doesn't seem like a good idea. I could document the regexp better though.

Comment (norm) [reply scripts/ci/runners/sync_authors.py:57]: (done, sort-of)

**Review by ashb — APPROVED** [review]

**Pull request merged**
